src/Nancy/Json/JavaScriptSerializer.cs

```diff
@@ -29,57 +29,40 @@
 namespace Nancy.Json
 {
     using System;
-    using System.Collections;
     using System.Collections.Generic;
-    using System.ComponentModel;
     using System.IO;
-    using System.Linq;
-    using System.Reflection;
-    using System.Text;
-
-    using Nancy.Helpers;
+    using Nancy.Json.Simple;
 
     public class JavaScriptSerializer
     {
         internal const string SerializedTypeNameKey = "__type";
 
-        List<IEnumerable<JavaScriptConverter>> _converterList;
-        List<IEnumerable<JavaScriptPrimitiveConverter>> _primitiveConverterList;
-
         int _maxJsonLength;
         int _recursionLimit;
         bool _retainCasing;
-        JavaScriptTypeResolver _typeResolver;
         bool _iso8601DateFormat;
 
+        readonly NancySerializationStrategy _serializerStrategy;
+
 #if NET_3_5
-        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(null, false, 2097152, 100);
+        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(false, 2097152, 100);
 
         public JavaScriptSerializer()
             : this(null, false, 2097152, 100)
         {
         }
-
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver)
-            : this(resolver, false, 2097152, 100)
-        {
-        }
 #else
-        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(null, false, 102400, 100, false, true, null, null);
+        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(false, 102400, 100, false, true, null, null);
 
         public JavaScriptSerializer()
-            : this(null, false, 102400, 100, false, true, null, null)
+            : this(false, 102400, 100, false, true, null, null)
         {
         }
 
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver)
-            : this(resolver, false, 102400, 100, false, true, null, null)
-        {
-        }
 #endif
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver, bool registerConverters, int maxJsonLength, int recursionLimit, bool retainCasing, bool iso8601DateFormat, IEnumerable<JavaScriptConverter> converters, IEnumerable<JavaScriptPrimitiveConverter> primitiveConverters)
+        public JavaScriptSerializer(bool registerConverters, int maxJsonLength, int recursionLimit, bool retainCasing, bool iso8601DateFormat, IEnumerable<JavaScriptConverter> converters, IEnumerable<JavaScriptPrimitiveConverter> primitiveConverters)
         {
-            _typeResolver = resolver;
+            _serializerStrategy = new NancySerializationStrategy(retainCasing, registerConverters, iso8601DateFormat, converters, primitiveConverters);

```

Comment:
Can we pass in an instance of the `JsonConfiguration` class here instead? @jchannon? Perhaps have an overload that takes the config object and calls this ctor? Not sure the greedy ctor would be used, this class won't be used outside anyway right?